File: client/src/app/models/author.ts
/**
 * Author model as returned by the API.
 */
export interface Author {
    name: string;
    sanitizedName: string;
    aliases: Array<string>;
    bio: string;
    
    /**
     * Array of books written by this author.
     */
    books: Array<{
        title: string;
        isbn: string;
        cover: string;
    }>;
}

File: java-server/src/main/java/com/mongodb/devrel/library/model/Author.java
package com.mongodb.devrel.library.model;

import org.bson.types.ObjectId;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Document(collection = "authors")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Author {
    @Id
    @Field("_id")
    private ObjectId id;

    private String name;

    private String sanitizedName;

    private String bio;

    private List<String> books;

    private List<String> aliases;
}
